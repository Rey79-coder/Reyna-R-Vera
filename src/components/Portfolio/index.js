import React from 'react';
import PhotoList from '../PhotoList';
import { capitalizeFirstLetter } from '../../utils/helpers';


function Portfolio({ currentCategory }) {
  
  const { name, description } = currentCategory;
  return (
    <section>
      <h1 data-testid="h1tag">{capitalizeFirstLetter(name)}</h1>
      <p>{description}</p>
      <PhotoList category={currentCategory.name} />
    </section>
  );
}

export default Portfolio;




/// IT'S WORKING WITH SINGLE PICTURE


// import React from 'react';
// import { capitalizeFirstLetter } from '../../utils/helpers';
// import photo from "../../assets/portfolio/0.png";


// function Portfolio(props) {

//     const currentCategory = {
//         name: "portfolio",
//         description:
//           "Screenshoots of the functional developed webpages",
//       };


//   return (
//     <section>
//       <h1>{capitalizeFirstLetter(currentCategory.name)}</h1>
//       <p>{currentCategory.name}</p>
//       <div>
//       <img
//             src={photo}
//             alt="Portfolio Samples"
//             className="img-thumbnail mx-1"
//           />
//       </div>

//     </section>
//   );
// }
// export default Portfolio;