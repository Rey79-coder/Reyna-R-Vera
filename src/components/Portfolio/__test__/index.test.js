import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Portfolio from ".."

const portfolio = { name: "portfolio", description: "Samples of the webpages I have developed" };

afterEach(cleanup)

describe('Portfolio component', () => {

  it('renders', () => {
    render(<Portfolio currentCategory={portfolio} />);
  });
})

it('matches snapshot', () => {
    const { asFragment } = render(<Portfolio currentCategory={portfolio} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders', () => {
    const { getByTestId } = render(<Portfolio currentCategory={portfolio} />)
    expect(getByTestId('h1tag')).toHaveTextContent('Portfolio')
  })