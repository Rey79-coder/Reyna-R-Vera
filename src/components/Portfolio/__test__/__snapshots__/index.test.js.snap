// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Portfolio
    </h1>
    <p>
      Samples of the webpages I have developed
    </p>
    <div>
      <div
        class="flex-row"
      >
        <img
          alt="echeriri.com"
          class="img-thumbnail mx-1"
        />
        <img
          alt="gineerblog"
          class="img-thumbnail mx-1"
        />
        <img
          alt="portfolio generator"
          class="img-thumbnail mx-1"
        />
        <img
          alt="pizza hunt"
          class="img-thumbnail mx-1"
        />
        <img
          alt="taskinator"
          class="img-thumbnail mx-1"
        />
        <img
          alt="echeriri.1.0"
          class="img-thumbnail mx-1"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
